### ISSUE TITLE
[PAY NATHAN] Reactions on  "settled up"/ unsettled messages doesn't sync across devices

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19394]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open account A on a desktop and on a mobile phone as well (2 different devices)
2. Open chat with account B on both devices
3. Make sure to have a settled or unsettled money request from account A to account B. (i.e there should be a history where account A sent a money request to account B and account B settled it) 
4. Send a test message from account A to account B
5. React on the last test message sent on one device and notice that the reaction is synced across both devices
6. React to the "settled up ..."/unsettled message on one device and observe the result on the other device

## Note:
It's not unique to settled money requests for for unsettled money requests as well.

## Expected Result:
Reactions on "settled up ..."/unsettled messages should be synced across devices just as it does on normal comments.

## Actual Result:
Reactions on "settled up ..."/unsettled messages are not synced across devices.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16.6

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/77a4cdfc-d140-4803-8ee8-aa57618084c0

https://github.com/Expensify/App/assets/93399543/c64122c5-2b60-428d-b8ed-087bbb722f55

https://github.com/Expensify/App/assets/93399543/575f1ba5-1da7-4c52-b9af-3ad06a092673

https://github.com/Expensify/App/assets/93399543/910ae44a-2b8f-4f57-afa3-93d7778f23e3

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Nathan-Mulugeta

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684405630552719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d43b2983d085915c</li>
        <li>Upwork Job ID: 1661008462238674944</li>
        <li>Last Price Increase: 2023-05-23</li>
    </ul>
</details>

**Comment by abekkala** [comment]
https://expensify.slack.com/archives/C049HHMV9SM/p1684787559710729?thread_ts=1684405630.552719&cid=C049HHMV9SM

**Comment by tienifr** [comment]
I think this is a back-end issue. Let's take a look at Add reaction API response:

- "Settle up" message:

![image](https://github.com/Expensify/App/assets/113963320/6c77b939-fd51-4fef-8496-2cae910fb281)

- Normal message:

![image](https://github.com/Expensify/App/assets/113963320/d09ecbe7-4e24-4a8e-b0d4-fb3a00e7bf72)

**Comment by abekkala** [comment]
@stitesExpensify what are your thoughts here?

**Comment by stitesExpensify** [comment]
I agree, it looks like this is probably an internal problem.

**Comment by stitesExpensify** [comment]
Will look into this tomorrow

**Comment by stitesExpensify** [comment]
I am actively working on this

**Comment by stitesExpensify** [comment]
i'm 99% sure the problem here is [this code](https://github.com/Expensify/Web-Expensify/blame/2e36d0cecb261acc4555a8c24ae51a7c4f55b4df/lib/ReportAPI.php#L686-L694). We are sending the response in the format of a `ADDCOMMENT` when it needs to be `REPORTPREVIEW`. The comment says it is temporary, and should be removed after [this PR](https://github.com/Expensify/Auth/pull/7782) is completed. Putting this on hold until then.

**Comment by abekkala** [comment]
@stitesExpensify we still good holding on this one?

**Comment by stitesExpensify** [comment]
Yes, that PR has still not been merged

**Comment by abekkala** [comment]
@jliexpensify Reassigning for Sabbatical. 

____

### CURRENT STATUS: 
This was put on hold as we're waiting on the PR to merge for another issue that may resolve this one. 

Issue reported by: @Nathan-Mulugeta
Fix: TBD, if needed. Fix for another issue may resolve this one
PR Review: TBD, if needed

**Comment by jliexpensify** [comment]
Not overdue, still seeking proposals

**Comment by jliexpensify** [comment]
On Hold

**Comment by jliexpensify** [comment]
Oh Hold

**Comment by stitesExpensify** [comment]
That issue has been merged, so we should try to reproduce this again and see if it's still happening

**Comment by jliexpensify** [comment]
Testing now!

**Comment by jliexpensify** [comment]
Payment Summary:

Bug Reporter: @Nathan-Mulugeta $250

NEW Upworks job

**Comment by jliexpensify** [comment]
Payment sorted, job closed.
